src/textual/renderables/sparkline.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import statistics
+from typing import Sequence, Iterable, Callable, TypeVar
+
+from rich.color import Color
+from rich.console import ConsoleOptions, Console, RenderResult
+from rich.segment import Segment
+from rich.style import Style
+
+T = TypeVar("T", int, float)
+
+
+class Sparkline:
+    """A sparkline representing a series of data.
+
+    Args:
+        data (Sequence[T]): The sequence of data to render.
+        width (int, optional): The width of the sparkline/the number of buckets to partition the data into.
+        min_color (Color, optional): The color of values equal to the min value in data.
+        max_color (Color, optional): The color of values equal to the max value in data.
+        summary_func (Callable[list[T]]): Function that will be applied to each bucket.
+    """
+
+    BARS = "▁▂▃▄▅▆▇█"
+
+    def __init__(
+        self,
+        data: Sequence[T],
+        *,
+        width: int | None,
+        min_color: Color = Color.from_rgb(0, 255, 0),
+        max_color: Color = Color.from_rgb(255, 0, 0),
+        summary_func: Callable[[list[T]], float] = max,
```

Comment:
Can we make this `summary_function`, I prefer to avoid abbreviations, except when well established.